include/statespace/StateSpace.hpp

```diff
@@ -148,6 +148,25 @@ class StateSpace {
         return true;
     }
 
+    /// Given input state, convert it to dynamic vector representing state.
+    /// Length, values of state depend on the derived class
+    ///
+    /// \param state Input state
+    /// \param[out] state_vector State vector
+    /// \return True if conversion successful; false otherwise
+    virtual bool get_state_vector(
```

Comment:
`state_to_vector`

Reply:
Actually I think this function should be in the `State` class: `virtual Eigen::VectorXd vector() const = 0;`